### PR TITLE
Version 0.2.0

### PR DESCRIPTION
**Pull request by TuggyBoat**
bot.py
-renamed bot object class to ModBot

constants.py
-fixed `get_guild()`, function is not async

database.py
-fixed search function to properly search database

-added `insert_infraction()` function

Helpers.py
-created Helpers.py

-added `create_thread`, `find_thread`, `display_infractions`, and `get_rule` functions

ModCommands.py
-added `InfractionReport` modal for future infraction reporting, only sends a message to channel as of now

-moved rule retrieval to Helpers.py

-added `warn` command, only displays infractions right now

-added `find_thread` command, gets an infraction thread by user id

-added `view_infractions` interaction, displays a user's infractions

-added `infraction_message` interaction, for future usage in message deletion and reporting, only calls for `InfractionReport` modal

### CODE REVIEW

File: ptn/modbot/botcommands/ModCommands.py [hunk 1]
```diff
@@ -84,34 +135,92 @@ async def sync(self, ctx):
...
-        guild = interaction.channel.guild
-        # get rule channel from guild
-        rules_channel_object = guild.get_channel(channel_rules())
-        # fetch rules message from rules channel
-        rules_message = await rules_channel_object.fetch_message(constants.rules_message())
+        # get member object
+        if member:
+            try:
+                member = interaction.guild.get_member(int(member))
+            except ValueError:
```
Comment (cimspin) [inline ptn/modbot/botcommands/ModCommands.py:149]: This should be moved to the Error Handler. We can create our own ValueError handler or just use CustomError, e.g.
```try:
    error = 'your error text'
    raise CustomError(error)
except Exception as e:
    return await on_generic_error(interaction, e)```
This is to enforce a consistent format for error reporting and logging.

File: ptn/modbot/botcommands/ModCommands.py [hunk 2]
```diff
@@ -84,34 +135,92 @@ async def sync(self, ctx):
...
-                    await interaction.response.send_message(f"Could not mention member. {e}", ephemeral=True)
-            await interaction.channel.send(embed=rules_list[rule_number - 1])
-            await interaction.response.send_message(f"Sent rule in {interaction.channel.name}", ephemeral=True)
-
-        except IndexError:
-            await interaction.response.send_message("That rule doesn't exist!", ephemeral=True)
\ No newline at end of file
+            member_object = guild.get_member(int(id))
+
+        except ValueError:
```
Comment (cimspin) [inline ptn/modbot/botcommands/ModCommands.py:181]: ➡️ Error Handler :]

File: ptn/modbot/botcommands/ModCommands.py [hunk 3]
```diff
@@ -84,34 +135,92 @@ async def sync(self, ctx):
...
+    await display_infractions(interaction=interaction, member=member, guild=guild)
+
+
+# An interaction to delete a violating message and send it to the infractions thread, with option to DM the user
+@bot.tree.context_menu(name='Infraction Message')
+@commands.has_any_role(*constants.any_elevated_role)
+async def infraction_message(interaction: discord.Interaction, message: discord.Message):
+    print(
+        f"infraction_message by {interaction.user.display_name} for user {message.author.display_name}'s message in {message.channel.id}.")
+    if message.author.bot:
```
Comment (cimspin) [inline ptn/modbot/botcommands/ModCommands.py:210]: This can also be shipped off to CustomError

File: ptn/modbot/database/database.py [hunk 4]
```diff
@@ -162,9 +162,16 @@ async def find_infraction(searchterm, searchcolumn):
     :returns: A list of InfractionData objects
     :rtype: InfractionData
     """
-    infraction_db.execute(
-        f"SELECT * FROM infractions WHERE {searchcolumn} LIKE (?)", (f'%{searchterm}%',)
-    )
+    if searchcolumn in [InfractionDbFields.entry_id.value, InfractionDbFields.warned_user.value,
+                        InfractionDbFields.warning_moderator.value]:
+        infraction_db.execute(
+            f"SELECT * FROM infractions WHERE {searchcolumn} = ?", (searchterm,)
```
Comment (cimspin) [inline ptn/modbot/database/database.py:168]: ♥

File: ptn/modbot/modules/Helpers.py [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+
+"""
+THREAD HELPERS
+
+Used for thread creation/interactions
+"""
+
+
+# create thread
+def create_thread(member: discord.Member, guild: discord.Member):
```
Comment (cimspin) [inline ptn/modbot/modules/Helpers.py:16]: `guild: discord.Guild`

File: ptn/modbot/modules/Helpers.py [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+            embed.description = embed.description + "\nUser has no infractions on record."
+        else:
+            for i, infraction in enumerate(infractions, start=1):
+                infraction_value = f'<t:{infraction.warning_time}:f> | Rule Broken: {infraction.rule_broken} | Warning ' \
+                                   f'Reason: {infraction.warning_reason}'
+                embed.add_field(name=f'Infraction {i}', value=infraction_value)
+
+        await interaction.response.send_message(embed=embed, ephemeral=True)
+
+    except Exception as e:
```
Comment (cimspin) [inline ptn/modbot/modules/Helpers.py:98]: ⋆｡  🎀  𝒯💙 𝓉𝒽𝑒 𝑒𝓇𝓇❀𝓇 𝒽𝒶𝓃𝒹𝓁𝑒𝓇 𝓅𝓁𝑒𝒶𝓈𝑒  🎀  ｡⋆

**Review by cimspin — CHANGES_REQUESTED** [review]
All great stuff. I can do the error handler bits if you like.

**Comment by TuggyBoat** [comment]
Did the Error handling work, all use CustomError now until we decide to use other methods

**Review by cimspin — APPROVED** [review]

**Pull request merged**